lib/middleware/request_tracker.rb

```diff
@@ -446,65 +477,56 @@ def rate_limit(request, cookie)
     return if @@ip_skipper&.call(ip)
     return if STATIC_IP_SKIPPER&.any? { |entry| entry.include?(ip) }
 
-    ip_or_id = ip
-    limit_on_id = false
-    if cookie && cookie[:user_id] && cookie[:trust_level] &&
-         cookie[:trust_level] >= GlobalSetting.skip_per_ip_rate_limit_trust_level
-      ip_or_id = cookie[:user_id]
-      limit_on_id = true
-    end
+    rate_limiter = self.class.rate_limiters_stack.active_rate_limiter(request, cookie)
+    return nil if rate_limiter.nil?
+    rate_limit_key = rate_limiter.rate_limit_key
```

Comment:
This is abit more complicated because we then have to consider the existing APIs for skipping rate limiters for an IP. I'm not opposed to it but I think we add this behaviour in another PR if there is a use case for it.

Reply:
My thought was we could deprecate/remove them which means outright removing code 👍

♥ removing code